service/matching/matchingEngine.go

```diff
@@ -689,16 +679,14 @@ func (e *matchingEngineImpl) QueryWorkflow(
 		return nil, err
 	}
 
-	shouldDrop, err := e.shouldDropTask(origTaskQueue, queryRequest.VersionDirective)
-	if err != nil {
-		return nil, err
-	} else if shouldDrop {
-		return nil, serviceerror.NewFailedPrecondition("Operations on versioned workflows are disabled")
-	}
 	// We don't need the userDataChanged channel here because we either do this sync (local or remote)
 	// or fail with a relatively short timeout.
 	taskQueue, _, err := e.redirectToVersionedQueueForAdd(ctx, origTaskQueue, queryRequest.VersionDirective, stickyInfo)
 	if err != nil {
+		if err == errUserDataDisabled {
+			// Rewrite to nicer error message
+			err = serviceerror.NewFailedPrecondition("Operations on versioned workflows are disabled")
```

Comment:
Just noting that the implication here is that this will crash Core based SDKs after a grace period of 1 minute.
I _think_ that's better than silently keeping those workers around idle.